### PR TITLE
[RFC] refactor(docs): move apidocs to `/docs`

### PR DESCRIPTION
**Pull request by raymondfeng**
This PR changes how api docs are generated and published.

1. Generate api docs for each package into `docs/api-docs/<pkg>`
  - Each package now has `docs/api-docs/<pkg-name>.html
  - No more duplicate assets for fonts/css/js
  - @loopback/docs itself servers as index.html for all packages
2. Publish api docs as part of `@loopback/docs`
3. Remove `api-docs` from each published packages to reduce bundle sizes (#1134)

We'll adjust apidocs.loopback.io site to use `@loopback/docs` for LoopBack 4 api docs.

## Checklist

- [x] `npm test` passes on your machine
- [ ] New tests added or existing tests modified to cover all changes
- [ ] Code conforms with the [style guide](http://loopback.io/doc/en/contrib/style-guide.html)
- [x] API Documentation in code was updated
- [x] Documentation in [/docs/site](../tree/master/docs/site) was updated
- [ ] Affected artifact templates in `packages/cli` were updated
- [ ] Affected example projects in `packages/example-*` were updated

### CODE REVIEW

File: packages/build/bin/generate-apidocs.js [hunk 1]
```diff
@@ -42,7 +42,13 @@ function run(argv, options) {
     args.push('--tsconfig', config);
   }
   if (!utils.isOptionSet(apidocsOpts, '--out', '-o')) {
-    args.push('-o', 'api-docs');
+    let out = 'api-docs';
+    if (process.env.LERNA_ROOT_PATH) {
```
Comment (b-admike) [inline packages/build/bin/generate-apidocs.js]: Is this dependent on #1189 landing?

Comment (bajtos) [inline packages/build/bin/generate-apidocs.js]: Could you please remind me: which code/script is setting LERNA_ROOT_PATH?

File: packages/openapi-spec-builder/package.json [hunk 2]
```diff
@@ -9,7 +9,7 @@
     "build": "lb-tsc es2017",
     "build:apidocs": "lb-apidocs",
     "clean": "lb-clean loopback-openapi-spec-builder*.tgz dist package api-docs",
-    "prepublishOnly": "npm run build && npm run build:apidocs",
+
```
Comment (bajtos) [inline packages/openapi-spec-builder/package.json]: Could you please remove these empty lines? They will be eventually removed by `npm` anyways.

**Comment by bajtos** [comment]
I am not able to fully consider ramifications of this change and whether it may break anything. If we agree this is the direction we want to take, then the code changes look mostly good to me - see my comments above.

File: packages/build/bin/generate-apidocs.js (hunk 1 above)
Comment (raymondfeng) [reply packages/build/bin/generate-apidocs.js]: yes

Comment (raymondfeng) [reply packages/build/bin/generate-apidocs.js]: See #1189

Comment (raymondfeng) [reply packages/build/bin/generate-apidocs.js]: #1189 is landed. I rebased the PR to master.

File: packages/example-hello-world/package.json [hunk 3]
```diff
@@ -12,7 +12,7 @@
     "build": "lb-tsc es2017",
     "build:apidocs": "lb-apidocs",
     "clean": "lb-clean *example-hello-world*.tgz dist package api-docs",
-    "prepublishOnly": "npm run build && npm run build:apidocs",
+
```
Comment (shimks) [inline packages/example-hello-world/package.json]: nitpick: no new line

File: packages/openapi-v3-types/package.json [hunk 4]
```diff
@@ -15,7 +15,7 @@
     "build": "lb-tsc es2017",
     "build:apidocs": "lb-apidocs",
     "clean": "lb-clean loopback-openapi-v3-types*.tgz dist package api-docs",
-    "prepublishOnly": "npm run build && npm run build:apidocs",
+
```
Comment (shimks) [inline packages/openapi-v3-types/package.json]: nitpick: no new line

Comment (raymondfeng) [reply packages/openapi-v3-types/package.json]: fixed

File: packages/example-hello-world/package.json (hunk 3 above)
Comment (raymondfeng) [reply packages/example-hello-world/package.json]: fixed

File: package.json [hunk 5]
```diff
@@ -40,7 +41,7 @@
     "build:full": "npm run clean:lerna && npm run bootstrap && npm test",
     "pretest": "npm run clean && npm run build",
     "test": "node packages/build/bin/run-nyc npm run mocha --scripts-prepend-node-path",
-    "mocha": "node packages/build/bin/run-mocha \"packages/*/DIST/test/**/*.js\" \"examples/*/DIST/test/**/*.js\" \"packages/cli/test/*.js\"",
+    "mocha": "node packages/build/bin/run-mocha \"packages/*/DIST/test/**/*.js\" \"examples/*/DIST/test/**/*.js\" \"packages/cli/test/*.js\" \"packages/build/test/integration/*.js\"",
```
Comment (bajtos) [inline package.json]: Would you mind making this more future-proof by using `\"packages/build/test/*/*.js\"`? I.e. allow any test category, not only `integration`. Unless I am missing something and the wildcard won't work.

Comment (raymondfeng) [reply package.json]: I wanted to exclude `packages/build/test/fixtures`. We can move `fixtures` to be inside `integration`.

**Pull request merged**